tracer/src/Datadog.Trace/DuckTyping/DuckType.Statics.cs

```diff
@@ -30,86 +31,112 @@ public static partial class DuckType
+        private static readonly PropertyInfo? _duckTypeInstancePropertyInfo;
+        [DebuggerBrowsable(DebuggerBrowsableState.Never)]
+        private static readonly MethodInfo? _methodBuilderGetToken;
+        [DebuggerBrowsable(DebuggerBrowsableState.Never)]
+        private static readonly ConstructorInfo? _ignoresAccessChecksToAttributeCtor;
 
         [DebuggerBrowsable(DebuggerBrowsableState.Never)]
         private static long _assemblyCount;
         [DebuggerBrowsable(DebuggerBrowsableState.Never)]
         private static long _typeCount;
 
-        /// <summary>
-        /// Gets the Type.GetTypeFromHandle method info
-        /// </summary>
-        public static readonly MethodInfo GetTypeFromHandleMethodInfo;
-
-        /// <summary>
-        /// Gets the Enum.ToObject method info
-        /// </summary>
-        public static readonly MethodInfo EnumToObjectMethodInfo;
-
         static DuckType()
         {
             Locker = new();
             DuckTypeCache = new();
             ActiveBuilders = new();
             IgnoresAccessChecksToAssembliesSetDictionary = new();
 
-            if (typeof(Type).GetMethod(nameof(Type.GetTypeFromHandle)) is { } getTypeFromHandleMethodInfo)
-            {
-                GetTypeFromHandleMethodInfo = getTypeFromHandleMethodInfo;
-            }
-            else
-            {
-                DuckTypeException.Throw($"{nameof(Type)}.{nameof(Type.GetTypeFromHandle)}() cannot be found.");
-            }
+            _getTypeFromHandleMethodInfo = typeof(Type).GetMethod(nameof(Type.GetTypeFromHandle));
+            _enumToObjectMethodInfo = typeof(Enum).GetMethod(nameof(Enum.ToObject), new[] { typeof(Type), typeof(object) });
+            _duckTypeInstancePropertyInfo = typeof(IDuckType).GetProperty(nameof(IDuckType.Instance));
+            _methodBuilderGetToken = typeof(MethodBuilder).GetMethod("GetToken", BindingFlags.Public | BindingFlags.NonPublic | BindingFlags.Instance);
+            _ignoresAccessChecksToAttributeCtor = typeof(IgnoresAccessChecksToAttribute).GetConstructor(new[] { typeof(string) });
 
-            if (typeof(Enum).GetMethod(nameof(Enum.ToObject), new[] { typeof(Type), typeof(object) }) is { } enumToObjectMethodInfo)
-            {
-                EnumToObjectMethodInfo = enumToObjectMethodInfo;
-            }
-            else
-            {
-                DuckTypeException.Throw($"{nameof(Enum)}.{nameof(Enum.ToObject)}() cannot be found.");
-            }
+            _assemblyCount = 0;
+            _typeCount = 0;
+        }
 
-            if (typeof(IDuckType).GetProperty(nameof(IDuckType.Instance)) is { } duckTypeInstancePropertyInfo)
-            {
-                DuckTypeInstancePropertyInfo = duckTypeInstancePropertyInfo;
-            }
-            else
+        /// <summary>
+        /// Gets the Type.GetTypeFromHandle method info
+        /// </summary>
+        public static MethodInfo GetTypeFromHandleMethodInfo
```

Comment:
Random thought. You id some work to optimise _not_ having multiple `static` properties elsewhere, and instead having a single `static Instance` field, that contains multiple instance properties, because of performance reasons (on .NET FX IIRC?). Is that something that would be worth doing here too do you think? 🤔 Not for this PR obviously

Reply:
Oh, I like the idea!